elrond-wasm/src/storage/mappers/linked_list_mapper.rs

```diff
@@ -0,0 +1,240 @@
+		self.set_u32(LEN_IDENTIFIER, len);
+	}
+
+	fn generate_new_key(&self) -> u32 {
+		let new_key = self.get_u32(NEW_KEY_IDENTIFIER) + 1;
+		self.set_u32(NEW_KEY_IDENTIFIER, new_key);
+		new_key
+	}
+
+	fn get_front(&self) -> u32 {
+		self.get_u32(FRONT_IDENTIFIER)
+	}
+
+	fn set_front(&self, index : u32) {
+		self.set_u32(FRONT_IDENTIFIER, index);
+	}
+
+	fn get_back(&self) -> u32 {
+		self.get_u32(BACK_IDENTIFIER)
+	}
+
+	fn set_back(&self, index : u32) {
+		self.set_u32(BACK_IDENTIFIER, index);
+	}
+
+	fn get_node(&self, index : u32) -> Node {
+		storage_get(self.api.clone(), self.build_index_named_key(NODE_IDENTIFIER, index).as_slice())
+	}
+
+	fn set_node(&self, index : u32, item : Node) {
+		storage_set(self.api.clone(), self.build_index_named_key(NODE_IDENTIFIER, index).as_slice(), &item);
+	}
+
+	fn clear_node(&self, index : u32) {
+		storage_set(self.api.clone(), self.build_index_named_key(NODE_IDENTIFIER, index).as_slice(), &BoxedBytes::empty());
+	}
+
+	fn get_value(&self, index : u32) -> T {
+		storage_get(self.api.clone(), self.build_index_named_key(VALUE_IDENTIFIER, index).as_slice())
+	}
+
+	fn set_value(&self, index : u32, value : T) {
+		storage_set(self.api.clone(), self.build_index_named_key(VALUE_IDENTIFIER, index).as_slice(), &value)
+	}
+
+	fn clear_value(&self, index : u32) {
+		storage_set(self.api.clone(), self.build_index_named_key(VALUE_IDENTIFIER, index).as_slice(), &BoxedBytes::empty())
+	}
+
+	pub fn is_empty(&self) -> bool {
+		self.get_len() == 0
+	}
+
+	pub fn len(&self) -> usize {
+		self.get_len() as usize
+	}
+
+	pub fn push_back(&mut self, elt: T) {
+		let new_key = self.generate_new_key();
+		let len = self.get_len();
+		let mut previous = NULL_ENTRY;
+		if len == 0 {
+			self.set_front(new_key);
+		} else {
+			let back = self.get_back();
+			let mut back_node = self.get_node(back);
+			back_node.next = new_key;
+			previous = back;
+			self.set_node(back, back_node);
+		}
+		self.set_node(new_key, Node {
+			previous,
+			next : NULL_ENTRY
+		});
+		self.set_back(new_key);
+		self.set_value(new_key, elt);
+		self.set_len(len + 1);
+	}
+
+	pub fn front(&self) -> Option<T> {
+		let front = self.get_front();
+		if front == NULL_ENTRY {
+			return None;
+		}
+		return Some(self.get_value(front));
+	}
+
+	pub fn remove_at_index(&self, index : u32) {
+		let len = self.get_len();
+		if len == 0 {
+			return;
+		}
+		let node = self.get_node(index);
+
+		if node.previous == NULL_ENTRY {
+			self.set_front(node.next);
+		} else {
+			let mut previous = self.get_node(node.previous);
+			previous.next = node.next;
+			self.set_node(node.previous, previous);
+		}
+
+		if node.next == NULL_ENTRY {
+			self.set_back(node.previous);
+		} else {
+			let mut next = self.get_node(node.next);
+			next.previous = node.previous;
+			self.set_node(node.next, next);
+		}
+
+		self.clear_node(index);
+		self.clear_value(index);
+		self.set_len(len - 1);
+	}
+
+	#[inline]
+    pub fn iter(&self) -> Iter<SA, T> {
```

Comment:
please call `cargo fmt` to nicely format the code

Reply:
I enabled the option from VS code to always format on file save. Done.